consumer/src/main/kotlin/streams/kafka/KafkaEventSink.kt

```diff
@@ -52,54 +55,56 @@ class KafkaEventSink(private val config: StreamsConfig,
             "schemaRegistryUrl" to "kafka.schema.registry.url",
             "groupId" to "kafka.${ConsumerConfig.GROUP_ID_CONFIG}")
 
-    private lateinit var errorService: ErrorService
-
     override fun getEventConsumerFactory(): StreamsEventConsumerFactory {
         return object: StreamsEventConsumerFactory() {
             override fun createStreamsEventConsumer(config: StreamsConfig, log: Log): StreamsEventConsumer {
                 val kafkaConfig = KafkaSinkConfiguration.from(config, db.databaseName())
                 return if (kafkaConfig.enableAutoCommit) {
-                    KafkaAutoCommitEventConsumer(kafkaConfig, log, getErrorService())
+                    KafkaAutoCommitEventConsumer(kafkaConfig, log)
                 } else {
-                    KafkaManualCommitEventConsumer(kafkaConfig, log, getErrorService())
+                    KafkaManualCommitEventConsumer(kafkaConfig, log)
                 }
             }
         }
     }
 
-    private fun getErrorService(): ErrorService {
-        if (!this::errorService.isInitialized) {
-            val kafkaConfig = KafkaSinkConfiguration.create(config, db.databaseName())
-            this.errorService = KafkaErrorService(kafkaConfig.asProperties(),
-                    ErrorService.ErrorConfig.from(kafkaConfig.streamsSinkConfiguration.errorConfig),
-                    { s, e -> log.error(s,e as Throwable) })
-        }
-        return this.errorService
-    }
-
-    override fun start() { // TODO move to the abstract class
-        if (StreamsPluginStatus.RUNNING == status()) {
+    override fun start() = runBlocking { // TODO move to the abstract class
```

Comment:
with stopping/restarting, probably need check for systemdb availability here, since it can go wrong & become unavailable during normal operations.

Reply:
Can it happen that with the database up&running the systemdb is not available?